packages/picasso.js/src/web/renderer/canvas-renderer/canvas-renderer.js

```diff
@@ -277,7 +280,14 @@ export function renderer(sceneFn = sceneFactory) {
     }
 
     hasChangedRect = false;
-    scene = newScene;
+    if (typeof settings.progressive !== 'function' || !settings.progressive()) {
+      scene = newScene;
+      scene.shapes = shapes;
+    } else {
+      const combinedShapes = [...scene.shapes, ...shapes];
+      scene = canvasRenderer.getScene(combinedShapes);
+      scene.shapes = combinedShapes;
```

Comment:
Is it nesessary to recreate the scene will all the current shapes? (instead of doing it with only the new nodes and combine the results)
or is the performace cost of redoing the scene for the same not small enought to not be a problem.

Reply:
Yeah, I agree, this doesn't feel optimal...

Reply:
This is safer and more correct otherwise we have to find out where children nodes to store in a scene and which can be different depending on components.